### PR TITLE
Add ga4 tracking to chat entry component

### PR DESCRIPTION
**Pull request by JamesCGDS**
## What
This PR adds GA4 tracking to the chat entry component.

## Why
Requested by the PA on the AI team.

### CODE REVIEW

File: app/views/govuk_publishing_components/components/docs/chat_entry.yml [hunk 1]
```diff
@@ -34,3 +34,6 @@ examples:
   with_margin_bottom:
     data:
       margin_bottom: 3
+  without_ga4_tracking:
+    data:
+      disable_ga4: true
```
Comment (AshGDS) [inline app/views/govuk_publishing_components/components/docs/chat_entry.yml:39]: I think you should add the description to this like so https://github.com/alphagov/govuk_publishing_components/blob/7372183409497ce80237f0ed950bde9b697b3db2/app/views/govuk_publishing_components/components/docs/back_link.yml#L20

File: app/views/govuk_publishing_components/components/_chat_entry.html.erb [hunk 2]
```diff
@@ -7,12 +7,23 @@
...
+  component_helper.add_data_attribute({ module: "ga4-link-tracker" }) unless disable_ga4
+
+  unless disable_ga4
+    ga4_link_data = {}
+    ga4_link_data[:ga4_link] = {
+      event_name: "navigation",
+      type: "callout",
+      section: "GOV.UK Chat entry",
+  }.to_json
+  end
```
Comment (AshGDS) [inline app/views/govuk_publishing_components/components/_chat_entry.html.erb:26]: Can you put the `ga4_link` straight into the object? E.g.

```ruby
ga4_link_data = {
  ga4_link: {
    # attributes in here
  }
}.to_json
```

**Review by AshGDS — CHANGES_REQUESTED** [review]
Looks good 👍  Have some minor suggestions, and this will also need a changelog entry.

File: app/views/govuk_publishing_components/components/docs/chat_entry.yml (hunk 1 above)
Comment (JamesCGDS) [reply app/views/govuk_publishing_components/components/docs/chat_entry.yml:41]: Thanks for spotting that! Have updated

File: app/views/govuk_publishing_components/components/_chat_entry.html.erb (hunk 2 above)
Comment (JamesCGDS) [reply app/views/govuk_publishing_components/components/_chat_entry.html.erb:27]: Cool, have updated

**Comment by JamesCGDS** [comment]
Thanks for the review @AshGDS - have made those changes now 👍

**Review by AshGDS — APPROVED** [review]
LGTM 👍

**Pull request merged**